dds_cli/account_manager.py

```diff
@@ -120,121 +91,52 @@ def delete_user(self, email, is_invite: bool = False):
+            headers=self.token,
+            json=json,
+            error_message="Failed to delete user",
+        )
 
-            # Get response
-            response_json = response.json()
-            message = response_json["message"]
-
-        except requests.exceptions.RequestException as err:
-            raise dds_cli.exceptions.ApiRequestError(
-                message=(
-                    "Failed to delete user"
-                    + (
-                        ": The database seems to be down."
-                        if isinstance(err, requests.exceptions.ConnectionError)
-                        else "."
-                    )
-                )
-            )
-        except simplejson.JSONDecodeError as err:
-            raise dds_cli.exceptions.ApiResponseError(message=str(err))
-
-        # Format response message
-        if not response.ok:
-            if response.status_code == http.HTTPStatus.INTERNAL_SERVER_ERROR:
-                raise dds_cli.exceptions.ApiResponseError(message)
-            else:
-                raise dds_cli.exceptions.DDSCLIException(message)
-        else:
-            LOG.info(message)
+        # Get response message
+        message = response_json["message"]
+
+        LOG.info(message)
 
     def delete_own_account(self):
         """Delete users from the system."""
         # Perform request to API
+        response_json = dds_cli.utils.perform_request(
+            dds_cli.DDSEndpoint.USER_DELETE_SELF,
+            method="delete",
+            headers=self.token,
+            json=None,
+            error_message="Failed to request deletion of account",
+        )
 
-        try:
-            response = requests.delete(
-                dds_cli.DDSEndpoint.USER_DELETE_SELF,
-                headers=self.token,
-                json=None,
-            )
-
-            # Get response
-            response_json = response.json()
-            message = response_json["message"]
-            dds_cli.auth.Auth.logout(self)
-
-        except requests.exceptions.RequestException as err:
-            raise dds_cli.exceptions.ApiRequestError(
-                message=(
-                    "Failed to request deletion of account"
-                    + (
-                        ": The database seems to be down."
-                        if isinstance(err, requests.exceptions.ConnectionError)
-                        else "."
-                    )
-                )
-            )
-        except simplejson.JSONDecodeError as err:
-            raise dds_cli.exceptions.ApiResponseError(message=str(err))
-
-        # Format response message
-        if not response.ok:
-            if response.status_code == http.HTTPStatus.INTERNAL_SERVER_ERROR:
-                raise dds_cli.exceptions.ApiResponseError(message)
-
-            raise dds_cli.exceptions.DDSCLIException(message)
+        # Get response message
+        message = response_json["message"]
+        dds_cli.auth.Auth.logout(self)
```

Comment:
Looks good. We can remove the json=None though, not sure why we've added it. Probably some test thing, for some reason.

Reply:
Added a suggestion for this below.